src/merkle/store/mod.rs

```diff
@@ -64,12 +64,16 @@ pub struct Node {
 /// }
 ///
 /// // Common internal nodes are shared, the two added trees have a total of 30, but the store has
-/// // only 10 new entries, corresponding to the 10 unique internal nodes of these trees.
-/// assert_eq!(store.num_internal_nodes() - 64, 10);
+/// // only 19 new entries.
+/// assert_eq!(store.num_nodes() - 64, 19);
 /// ```
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct MerkleStore {
-    nodes: BTreeMap<RpoDigest, Node>,
+    /// This map stores every node of every structure added to the store. When the value is set to
+    /// `None` the `key` represents a leaf node, if the value is [Node] then the key represent an
+    /// internal node with the specified `left` and `right` children, which may not necessarily be
+    /// in the store.
+    nodes: BTreeMap<RpoDigest, Option<Node>>,
```

Comment:
This isn't very intuitive maybe. It's clear we are intending to fallback to an empty root in case of `None`, but then maybe we should just write the empty root value directly?

Reply:
> This isn't very intuitive maybe. It's clear we are intending to fallback to an empty root in case of None, but then maybe we should just write the empty root value directly?

That is not the behavior here, the absence of a value indicates the key itself is a lead. The present of anything else would indicate the `key` is a parent node of whatever value is inside.